src/Features/Core/Portable/MoveToNamespace/AbstractMoveToNamespaceService.cs

```diff
@@ -0,0 +1,224 @@
+                    return MoveTypeToNamespace(analysisResult.Document, analysisResult.SyntaxNode, targetNamespace, cancellationToken);
+                default:
+                    throw new InvalidOperationException();
+            }
+        }
+
+        private static async Task<MoveToNamespaceResult> MoveItemsInNamespaceAsync(
+            Document document,
+            SyntaxNode container,
+            string targetNamespace,
+            CancellationToken cancellationToken)
+        {
+            var changeNamespaceService = document.GetLanguageService<IChangeNamespaceService>();
+            if (changeNamespaceService == null)
+            {
+                return MoveToNamespaceResult.Failed;
+            }
+
+            var changedSolution = await changeNamespaceService.ChangeNamespaceAsync(
+                document,
+                container,
+                targetNamespace,
+                cancellationToken).ConfigureAwait(false);
+
+            return new MoveToNamespaceResult(changedSolution, document.Id);
+        }
+
+        private static async Task<MoveToNamespaceResult> MoveTypeToNamespace(
+            Document document,
+            SyntaxNode container,
+            string targetNamespace,
+            CancellationToken cancellationToken)
+        {
+            var moveTypeService = document.GetLanguageService<IMoveTypeService>();
+            if (moveTypeService == null)
+            {
+                return MoveToNamespaceResult.Failed;
+            }
+
+            // The move service expects a single position, not a full selection
+            var moveSpan = new TextSpan(container.FullSpan.Start, 0);
+
+            var modifiedSolution = await moveTypeService.GetModifiedSolutionAsync(
+                document,
+                moveSpan,
+                MoveTypeOperationKind.MoveTypeNamespaceScope,
+                cancellationToken).ConfigureAwait(false);
+
+            var modifiedDocument = modifiedSolution.GetDocument(document.Id);
+            var syntaxRoot = await modifiedDocument.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            var syntaxNode = syntaxRoot.GetAnnotatedNodes(AbstractMoveTypeService.NamespaceScopeMovedAnnotation).FirstOrDefault();
+            if (syntaxNode == null)
+            {
+                syntaxNode = container.FirstAncestorOrSelf<TNamespaceDeclarationSyntax>();
+            }
+
+            return await MoveItemsInNamespaceAsync(
+                modifiedDocument,
+                syntaxNode,
+                targetNamespace,
+                cancellationToken).ConfigureAwait(false);
+        }
+
+        private static SymbolDisplayFormat QualifiedNamespaceFormat = new SymbolDisplayFormat(
+            globalNamespaceStyle: SymbolDisplayGlobalNamespaceStyle.Omitted,
+            typeQualificationStyle: SymbolDisplayTypeQualificationStyle.NameAndContainingTypesAndNamespaces);
```

Comment:
AFAICT this is only run on Namespaces, not types. Is this part unnecessary, or did I miss something?

Reply:
It's unnecessary to run on types. This only makes sure that FQN is presented to users for the available namespaces in a project.

Reply:
I meant the `typeQualificationStyle` bit. Is that irrelevant then?

Reply:
Yea, it's needed. It applies to `INamespaceOrTypeSymbol` afaict. I tried removing as well and it no longer gets the correct naming.